base/record_replay.h

```diff
@@ -43,6 +44,26 @@ struct AutoOrderedLock {
   int id_;
 };
 
+// Drop in replacement for base::Lock that can be used with
+// mixed 'auto' and 'try' lockers.
+class LOCKABLE ReplayBaseLock {
```

Comment:
I'm confused by why we need this class -- base::Lock supports ordering, why can't we just continue using that?

Reply:
I didn't realize that was the case!  Convenient, but not obvious from the signature.  I've observed that all of our changes involve replacing classes/functions with 'replay'-marked replacements, and even my searching for OrderedLock didn't turn up the changes to the base lock, so I never noticed it.  Regardless, not a big deal (the hard part was definitely _not_ writing the code,) so I've reverted the changes.